lib/data/data_source/trips_data_source_impl.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:injectable/injectable.dart';
+import 'package:trip_tally/data/api/api_client.dart';
+import 'package:trip_tally/data/dto/trips/add_trip_dto.dart';
+import 'package:trip_tally/domain/data_source/trips_data_source.dart';
+import 'package:trip_tally/domain/utils/exception.dart';
+import 'package:trip_tally/domain/utils/success.dart';
+import 'package:trip_tally/presentation/utils/enums/errors.dart';
+
+@Injectable(as: TripsDataSource)
+class TripsDataSourceImpl implements TripsDataSource {
+  TripsDataSourceImpl(this.client);
+
+  final ApiClient client;
+
+  @override
+  Future<Success> addTrip(AddTripDto dto) async {
```

Comment:
```suggestion
  Future<Success> createTrip(AddTripDto dto) async {
```
Because we use CRUD principles -> Create, read(get), update, delete